src/modules/nav/nav.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import { Router, NavigationStart } from '@angular/router';
+
+import { SkyAppConfig } from '@blackbaud/skyux-builder/runtime';
+
+import { StacheNavLink } from './nav-link';
+
+@Injectable()
+export class StacheNavService {
+  private activeRoutes: StacheNavLink[];
+
+  public constructor(
+    private appConfig: SkyAppConfig,
+    private router: Router
+  ) {
+    router.events.subscribe((val: any) => {
+      if (val instanceof NavigationStart) {
+        this.clearActiveRoutes();
+      }
+    });
+  }
+
+  public getActiveUrl(): string {
+    return this.router.url.split('#')[0];
+  }
+
+  public getActiveRoutes(): StacheNavLink[] {
+    if (this.activeRoutes) {
+      return this.clone(this.activeRoutes) as StacheNavLink[];
+    }
+
+    let rootPath = this.getActiveUrl().replace(/^\//, '').split('/')[0];
+    let routes = this.clone(this.appConfig.runtime.routes);
+
+    routes = routes.filter((route: any) => route.routePath.indexOf(rootPath) === 0);
+
+    routes = routes.map((route: any) => {
+      return {
+        segments: route.routePath.split('/'),
+        path: route.routePath
+      };
+    });
+
```

Comment:
This is a lot cleaner, though I would prefer not to chain off a method on our class.   

```
let clonedRoutes = this.clone(this.appConfig...)
let childRoutes = clonedRoutes.filter(....
```
Chain array methods off an array, not off the data returned from our method.  Especially since `this.clone` can return an object, not just an array.